### DISCUSSION TITLE
How to show default one row in repeaters?

### DISCUSSION THREAD

**Discussion by zeeshantariq08** [filamentphp/filament#12277] [Help] [answered]
### Package

Form builder

### Package Version

v3.0.0

### How can we help you?

@leandrocfe ?

``Tables\Actions\Action::make('Repeater')->form([
       Repeater::make('reviewers')
                            ->schema([

                                Select::make('reviewer_id')
                                    ->label('Reviewer')
                                    ->options(\App\Models\User::pluck('first_name', 'id'))
                                    ->required()
                                    ->distinct()
                                    ->searchable(),
                                TextInput::make('level')
                                    ->label('Level')
                                    ->required()
                                    ->numeric()
                                    ->minValue(10)
                                    ->step(10)
                            ])->addActionLabel('Add another reviewer')
                            ->columns(2)
                    ])``

**Comment by leandrocfe** [comment]
```php
->default([
    [
        'reviewer_id' => '1',
        'level' => '10'
    ],
])
```

**Reply by zeeshantariq08** [reply]
@leandrocfe it is not working, it is working in create or edit page button not in action form

**Reply by leandrocfe** [reply]
Share the whole code you are doing

**Reply by zeeshantariq08** [reply]
@leandrocfe please check
`<?php



class ModelsRelationManager extends RelationManager
{
   

    protected static string $relationship = 'models';


    protected static ?string $recordTitleAttribute = 'model';

    protected static bool $isLazy = false;

    public function form(Form $form): Form
    {
        return $form
            ->schema([
               //
            ]);
    }

    public function table(Table $table): Table
    {
        return $table
           
            ->filters([
                //
            ])
            ->headerActions([
        
            ])
            ->actions([
                Tables\Actions\Action::make('Create Reviewers Template')
                    ->label('Create Reviewers Template')
                    ->form([
                       

                        Repeater::make('reviewers')
                            ->schema([

                                Select::make('reviewer_id')
                                    ->label('Reviewer')
                                    ->options(\App\Models\User::all()->pluck('full_name', 'id')->toArray())
                                    ->required()
                                    ->distinct()
                                    ->searchable(),
                                TextInput::make('level')
                                    ->label('Level')
                                    ->required()
                                    ->numeric()
                                    ->minValue(10)
                                    ->step(10)
                            ])->addActionLabel('Add another reviewer')
                            ->columns(2)
                    ])
                    ->action(function (Tables\Actions\Action $action, array $data) {
                       
                        foreach ($data['reviewers'] as $reviewer) {
                            Reviewer::create([
                                'reviewer_id' => $reviewer['reviewer_id'],
                                'level' => $reviewer['level'],
                            ]);
                        }

                    
                    })

            
            
                })
            ])
            ->bulkActions([
                Tables\Actions\DeleteBulkAction::make(),
            ]);
    }


}
`

**Reply by leandrocfe** [reply]
Where is the default? Shouldn't it be this?

```php
Repeater::make('reviewers')
    ->default([
        [
            'reviewer_id' => '1',
            'level' => '10'
        ],
    ])
    ...
```

**Reply by zeeshantariq08** [reply]
@leandrocfe i tried this but nothing works and also tried defaultItems(1) but this either not working

**Reply by leandrocfe** [reply]
Weird. Are you able to share this project or a mini project to reproduce this issue, on github?

**Reply by zeeshantariq08** [reply]
hey @leandrocfe i found the way by using mountusing btw thanks for your help
```
->mountUsing(function (ComponentContainer $form, Model $record) {
    $data['reviewers'] = [
            ['reviewer_id' => '', 'level' => ''],
        ];
        $form->fill($data);

        return $form;
    })
```
```
->form([
    Repeater::make('reviewers')
        ->schema([

            Select::make('reviewer_id')
                ->label('Reviewer')
                ->options(\App\Models\User::all()->pluck('full_name', 'id')->toArray())
                ->required()
                ->distinct()
                ->searchable(),
            TextInput::make('level')
                ->label('Level')
                ->required()
                ->numeric()
                ->minValue(10)
                ->step(10)
        ])->addActionLabel('Add another reviewer')
        ->column
```s(2)
                ])

**Reply by leandrocfe** [reply, marked as answer]
I think you can do this, if you want to use it in the `->default()`

```php
->mountUsing(fn(ComponentContainer $form) => $form->fill())
```

**Reply by zeeshantariq08** [reply]
@leandrocfe thank you!
